presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -448,6 +448,7 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         Optional<Symbol> groupIdSymbol = Optional.empty();
         if (groupingSets.size() > 1) {
             groupIdSymbol = Optional.of(symbolAllocator.newSymbol("groupId", BIGINT));
+            analysis.addGroupIdSymbol(node, groupIdSymbol.get());
```

Comment:
See 15c59d02f5eb5c5bf87a8736a36bd03b3ae9323f. New code no longer modifies the `Analysis` object.